### ISSUE TITLE
General usage

### ISSUE THREAD

**Issue by jaumard** [Idean/Flutter-Neumorphic#121]
Here is just a general feedback after few days of use :)

I really like using this design system, but there is few things that I don't like:

- some naming are not consistent with flutter naming, example: `onClick` on button where it's `onPressed` on official designs like material or Cupertino.
- have to use material stuff
- have to always get theming colors to put in text `Text("...", style:TextStyle(color: NeumorphicTheme.defaultTextColor(context)),`
- sometime having some fields that are not supported (like #120) 

I understand that you don't want to redo everything material does. 
But I'm wondering if there is an hybrid solution where material stuff are hidden.

For example we can have a NeumorphicApp that under the hood setup a MaterialApp and compute the theming from a NeumorphicTheme, that will fix the need of putting `NeumorphicTheme.defaultTextColor ` everywhere and redefining theme twice (neumorphic + material).

That would be a good start :) 

Another things (but maybe a bit not necessary) would be to have `export 'package:flutter/material.dart' show TextField, Dialog...;` to give the illusion that neumorphic have it all ^^

Because when you use a design system you don't want to fill the blank with another one. 

My 2 cents guys ;) keep the good work!

**Comment by florent37** [comment]
Yes I already have the feeling that onPressed should be used instead of onClick

have to use material stuff : The flutter is so mixed with material, if you look inside cupertino for example, you have calls to material. So it's not usefull to duplicate all material class just for rename them "neumorphic.*"

**Comment by florent37** [comment]
I think i will rename onClick to `onPressed`

**Comment by jaumard** [comment]
> The flutter is so mixed with material, if you look inside cupertino for example, you have calls to material. So it's not usefull to duplicate all material class just for rename them "neumorphic.*"

You have it correctly by saying `look inside cupertino ` keyword being `inside`, you can do a full flutter app in Cupertino without referencing material, even if it's used under the hood by Cupertino.

My point being that users want to use Neumorphic don't want to have to play with material too, so as I say my idea is not to duplicate material class but more hiding the most used to avoid complexity using two different design system. 

Just by reexporting material classes with `'package:flutter/material.dart' show TextField, Dialog...;` will be enough to me as they will be available under neumorphic without any additional code to maintain on your side (if I'm not wrong, not sure ^^). 

But if you really want to push this design (and package) it will definitely be needed as for new comers it look half finish that way (only my opinion ^^) and will look complex to use too (as if something is not available go pick on material to fill the blank).

Don't get me wrong I love what you did here ! It's just the next steps that need to be done imho

**Comment by florent37** [comment]
look at material/theme.dart

```

import 'package:flutter/cupertino.dart';
import 'package:flutter/foundation.dart';
import 'package:flutter/widgets.dart';

```

material depends on cupertino :D

**Comment by florent37** [comment]
I tried to duplicate application & pages but it's just a copy paste useless, it does not add anything and if flutter change anything I will have to maintain the difference and make an evolution

**Comment by jaumard** [comment]
I'll try some things on my end and I'll do a proposal maybe it will be easier to visualize :)

**Comment by florent37** [comment]
for neumorphic App I tried here : https://github.com/Idean/Flutter-Neumorphic/pull/106

but you have to create neumorphic navigation, route, etc :'(

**Comment by jaumard** [comment]
Yep if you do like this you have to redo pretty much everything, but that's not my proposal.
My proposal is just to embed the MaterialApp inside a NeumorphicApp and compute automatically MaterialTheme from NeumorphicTheme. Like this need to to redo everything. I'll try to do a POC when I have some time. But pretty sure it should give something pretty decent to use ;)

**Comment by florent37** [comment]
if you want :) I tried but it's look too much :D

**Comment by florent37** [comment]
I'm sure you can do something great with neumorphic app & also your toolbar #122 
Can you try to add this in a fork then create a pull Request ? 

I'll be happy to count you as a contributor 🥰✌️

**Comment by florent37** [comment]
I published a `2.2.0` with the `testStyle` change, & `onPressed`

**Comment by florent37** [comment]
I set it to close, but I keep your issues in my mind, and will try to make the project more usable (scaffold, application, themes (with text) etc)

thanks !
